mw/i2c_service/controller/adcsensor/controller.hpp

```diff
@@ -0,0 +1,60 @@
+/**
+ * @file controller.hpp
+ * @author Mateusz Krajewski (matikrajek42@gmail.com)
+ * @brief 
+ * @version 0.1
+ * @date 2024-05-19
+ * 
+ * @copyright Copyright (c) 2024
+ * 
+ */
+
+#ifndef MW_I2C_SERVICE_CONTROLLER_ADCSENSOR_CONTROLLER_HPP_
+#define MW_I2C_SERVICE_CONTROLLER_ADCSENSOR_CONTROLLER_HPP_
+
+#include <unordered_map>
+#include <string>
+#include <vector>
+#include <future> // NOLINT
+
+#include "mw/i2c_service/controller/ads7828/controller.hpp"
+#include "core/json/json_parser.h"
+namespace simba {
+namespace i2c {
+
+struct SensorConfig {
+  uint8_t channel;
+  uint8_t a;
+  uint8_t b;
+};
+
+
+class ADCSensorController {
```

Comment:
to chyba jest final ?

Reply:
Tak

Reply:
Tylko on nie dziedziczy po adcController wiec potrzebny tu jest final?